New blueprint: use wp cli to add a post with a featured image

bph opened the pull request:
The goal: 

- use the content from the markdown file 
- create a post
- attach an image
- display it as landing page. 

This way one could create a playground instance with some test instructions right on the site. 
There are a few challenges, I haven't figured out. 
The write file step I tried both /wordpress/wp-content/uploads and /wp-content/uploads 

Error message: 
`config-cb0d0260.js:631 Error: Could not write to "/wp-content/uploads/postcontent.md": There is no such file or directory OR the parent directory does not exist.`


And I am not sure how to point wp-cli to the file to grab it and write it in a post. 

The image part is also tricky. 

Locally, I can use this 
```- wp media import ~/azblueprints/blueprints/wpcli-markdown/Select-storage-method.png --post_id=56 --title="Select your storage method" --featured_image```

WordPress then created the media library directory, placed the file with all the image sizes and created db attachment post.

bph committed:
v1 create post from markdown file with an image

bph committed:
v1 blueprint changes

bph committed:
added absolute paths

bph committed:
Just handle the md file.

adamziel commented on blueprints/wpcli-markdown/blueprint.json:
```diff
@@ -0,0 +1,27 @@
+{
+	"$schema": "https://playground.wordpress.net/blueprint-schema.json",
+	"meta": {
+		"title": "Content via wp-cli and markdown",
+		"description": "create post from markdown file with an image",
+		"author": "bph",
+		"categories": ["Content", "wpcli"]
+	},
+	"landingPage": "/wp-admin/",
+	"steps": [
+		{
+			"step": "login"
+		},
+		{
+			"step": "writeFile",
+			"path": "/wp-content/uploads/postcontent.md",
+			"data": {
+				"resource": "url",
+				"url": "https://gutenbergtimes.com/wp-content/uploads/2020/11/wpcli-markdown/postcontent.md"
+			}
+		},
+		{
+			"step": "wp-cli",
+			"command": "wp post create --post_title 'Welcome to Playground'  /wp-content/uploads/postcontent.md"
+		}
+	]
+}
```
Three problems here:

* It's `/wordpress/wp-content`, not `/wp-content`.
* `/wp-content/uploads` doesn't exist by default in WordPress, I updated the path to just `/wp-content/postcontent.md`
* `=` is needed in `wp post create --post_title='Welcome to Playground'`

```suggestion
{
	"$schema": "https://playground.wordpress.net/blueprint-schema.json",
	"meta": {
		"title": "Content via wp-cli and markdown",
		"description": "create post from markdown file with an image",
		"author": "bph",
		"categories": ["Content", "wpcli"]
	},
	"landingPage": "/wp-admin/",
	"steps": [
		{
			"step": "login"
		},
		{
			"step": "writeFile",
			"path": "/wordpress/wp-content/postcontent.md",
			"data": {
				"resource": "url",
				"url": "https://gutenbergtimes.com/wp-content/uploads/2020/11/wpcli-markdown/postcontent.md"
			}
		},
		{
			"step": "wp-cli",
			"command": "wp post create --post_title='Welcome to Playground' /wordpress/wp-content/postcontent.md"
		}
	]
}
```

bph committed:
Fix name of directories and  wp cli command

Co-authored-by: Adam Zielinski <adam@adamziel.com>

bph committed:
added media, switch content to blocks

bph committed:
use raw.githubusercontent.com as source

bph commented:
@adamziel Thank you so much for the correction of directories and the wp-cli command. 
I changed the content from mark down to block notation as WordPress wouldn't convert the markdown. 
And I added a wp-cli media command to add at featured image to the post. So cool that it works! 

To test it before the merge [use this link](https://playground.wordpress.net/?blueprint-url=https://gutenbergtimes.com/wp-content/uploads/2020/11/wpcli-markdown/blueprint.json)

bph commented:
Here is a screenshot of the post created: 
![Screenshot 2024-04-17 at 18 12 05](https://github.com/adamziel/blueprints/assets/39980/6aeb547b-bde4-4e9f-827a-4bb979ca2be7)

bph commented:
Nice going with the validity check on the PR! 
It identifies a non-allowed directory (gutenbergtimes.com site) and suggested `https://raw.githubusercontent.com/adamziel/blueprints/cli-markdown/` I added it but but that doesn't work... 
That's why I added the test link in an earlier comment.

bph committed:
update directory to use trunk branch

bph commented:
Updated the directories to use 'https://raw.githubusercontent.com/adamziel/blueprints/trunk/blueprints/'

bph commented:
Not quite sure how to solve this this failing check: 
![Screenshot 2024-04-18 at 08 27 06](https://github.com/adamziel/blueprints/assets/39980/8b8ca056-d638-44aa-be37-3ab3dda63abe)

adamziel commented:
@bph The problem with the https://raw.githubusercontent.com/adamziel/blueprints/trunk/blueprints/cli-markdown/postcontent.md URL is that it will work after merging, but it makes previewing the PR impossible as there isn't anything at that URL yet – it just return 404. 

The URL pointing to your branch, https://raw.githubusercontent.com/adamziel/blueprints/cli-markdown/blueprints/wpcli-markdown/postcontent.md, returns the correct file. The problem now is – once we merge this PR and delete the branch, it will start returning 404.

I've reconciled this dualism recently with a repository automation that rewrites branch-specific URLs to trunk URLs. The automation verifies your PR only contains URLs pointing to your branch, and once we merge, it will update your Blueprint to point to trunk. I also documented this in [CONTRIBUTING.md](https://github.com/adamziel/blueprints/blob/trunk/CONTRIBUTING.md).

LMK what do you think, I haven't found a better solution yet.

adamziel commented on blueprints/wpcli-markdown/blueprint.json:
```diff
@@ -0,0 +1,39 @@
+			"path": "/wordpress/wp-content/postcontent.md",
+			"data": {
+				"resource": "url",
+				"url": "https://raw.githubusercontent.com/adamziel/blueprints/trunk/blueprints/cli-markdown/postcontent.md"
```
```suggestion
				"url": "https://raw.githubusercontent.com/adamziel/blueprints/cli-markdown/blueprints/wpcli-markdown/postcontent.md"
```

adamziel commented on blueprints/wpcli-markdown/blueprint.json:
```diff
@@ -0,0 +1,39 @@
+			"path": "/wordpress/wp-content/Select-storage-method.png",
+			"data": {
+				"resource": "url",
+				"url": "https://raw.githubusercontent.com/adamziel/blueprints/trunk/blueprints/cli-markdown/Select-storage-method.png"
```
```suggestion
				"url": "https://raw.githubusercontent.com/adamziel/blueprints/cli-markdown/blueprints/wpcli-markdown/Select-storage-method.png"
```

adamziel committed:
Update blueprints/wpcli-markdown/blueprint.json

adamziel committed:
Update blueprints/wpcli-markdown/blueprint.json

adamziel commented:
I went ahead and updated the URLs in https://github.com/adamziel/blueprints/pull/27/commits/30fb07d1043565c992450205583448859044b7fa and https://github.com/adamziel/blueprints/pull/27/commits/65e40ef941724dadb6fb55ecd0967c4702cc7938. All checks are now green and the preview URL works.

adamziel commented:
I just confirmed that, after merging, the preview link in the Gallery document works as expected:

https://github.com/adamziel/blueprints/blob/trunk/GALLERY.md
